app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -143,6 +143,50 @@ public function syncProductReview(array $product_review): array {
         }
     }
 
+
+    public function hasConsent(Mage_Sales_Model_Order $order): bool {
+        $permission_url = 'https://dashboard.webwinkelkeur.nl/api/2.0/order_permissions.json?' .
+            http_build_query([
+                'orderNumber' => $order->getIncrementId(),
+                'id' => $this->getShopId(),
+                'code' => $this->getApiKey(),
+            ]);
+
+        $start_time = microtime(true);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, CURLOPT_URL, $permission_url);
+        try {
+            $response = curl_exec($ch);
+            $message = '';
+            if ($response === false) {
+                $message = curl_error($ch);
+            }
+            if ($response && !json_decode($response)->has_consent) {
+                $message = 'Invitation has not been sent as the customer did not consent.';
+            }
+            if (!empty($message)) {
```

Comment:
Instead of relying on the error message being empty, you should have an `else` with `return true;` after all the error conditions, and then put the logging below that `else`.